docs/high_availability_installation_guide.md

```diff
@@ -0,0 +1,284 @@
+#    } 
+#  } 
+#} 
+```
+    
+**Important!**:  
+
+You need to change the **VIP** to the real IP address. 
+Change **harbor_node1_IP**  and **harbor_node2_IP** to the real harbor nodes IPs. 
+If you have more than two nodes then please add more real_server definitions in the keepalived.conf 
+
+ - 3> Restart the Keepalived service to make the change take affect.  
+
+```
+#>systemctl keepalived restart
+```
+
+#### On Loadbalancer02: 
+
+Follow the same steps 1 to 3 as Loadbalancer01 list, only change the ```priority``` to 20 in the /etc/keepalived/keepalived.conf in step 2.  The higher number will get the VIP address.  
+
+#### Harbor node 1 setup 
+ - Download harbor offline package from GitHub to you home directory 
+ - Extract the harbor-offline-installer-v.x.x.x.tgz You will get a folder ```harbor``` in the current directory 
+ - cd to ```harbor``` directory 
+ - Edit the ```harbor.cfg``` file 
+ - Edit the hostname = reg.mydomain.com , change reg.mydomain.com to your FQDN or VIP 
+ - Edit the ```db_host``` to point to the external db host IP/FQDN 
+ - Edit the ```db_password``` to the password of the external DB 
+ - Edit the ```db_user``` to the user of the external DB
+ - Edit the ```db_port``` to the port of the external DB
+ - Edit the ```redis_url``` to the redis server or cluster address
+ - Edit the ```clair_db_host```, ```clair_db_port```, ```clair_db_username```, ```clair_db_password```, ```clair_db``` 
+ - Edit the ```registry_storage_provider_name``` and its related settings ```registry_storage_provider_config``` 
+You can find the configure examples for all kinds of storages at https://docs.docker.com/registry/configuration/#storage .  
+For example if you use swfit as the storage backend then you need to set the follow:
+
+```
+registry_storage_provider_name = swfit
+registry_storage_provider_config = username:yourusername,password: yourpass,authurl: http://192.168.1.217/identity/v3,tenant: admin,domain: default,region: RegionOne,container: docker_images
+```
+
+ - Import Harbor database schema to your external MariaDB 
+```
+    Copy the ~/harbor/ha/registry.sql file to a machine that has mysql client installed 
+    #> mysql -u your_db_username -p -h your_db_ip < registry.sql 
+```
+ -  (Optional) If you enable https, you need to prepare the certificate and key and copy them to /data/cert/ directory. You need to rename them to "server.key" and "server.crt". If you want keep your own filename for the certificate, you need to modify the ssl_cert and ssl_cert_key properties in harbor.cfg. If you use a certificate that signed by a private CA then you need to put your CA file to the /data/ca_download/ca.crt
+
+ -  Start the Harbor on the first node by run 
+```
+#>./install.sh --ha 
+```
+**Note!** If you want to use vulnerability scanning functionality. Then use the follow command instead
+
+```
+          #>./install.sh --ha --with-clair 
+```
+
+ - Zip the harbor directory 
+
+```
+tar - cvf harbor_ha.tar ~/harbor 
```

Comment:
if prefer to keep "#>", then we can use "#" for all scripts running as root and "$" for all scripts running as non-root

Reply:
We will use #> command prompt for all command.

Reply:
'#>' as the command prompt for user root is seldom seen in docs. '# ' is used or no '#' prompt. If keeping '#' then add a space after '#', e.g. '# ./install.sh ' instead of '#>./install.sh' or '#./install.sh'.